lib/msal-core/src/ErrorMessage.ts

```diff
@@ -21,13 +21,127 @@
+  },
+  tokenRenewalError: {
+    code: "token_renewal_error",
+    desc: "Token renewal operation failed due to timeout"
+  },
+  invalidStateError: {
+    code: "invalid_state_error",
+    desc: "Invalid state"
+  },
+  nonceMismatchError: {
+    code: "nonce_mismatch_error",
+    desc: "Nonce is not matching, Nonce received: "
+  },
+  loginProgressError: {
+    code: "login_progress_error",
+    desc: "Login_In_Progress: Error during login call - login is already in progress"
+  },
+  acquireTokenProgressError: {
+    code: "acquiretoken_progress_error",
+    desc: "AcquireToken_In_Progress: Error during login call - login is already in progress."
+  },
+  userCancelledError: {
+    code: "user_cancelled",
+    desc: "User cancelled the flow"
+  },
+  callbackError: {
+    code: "callback_error",
+    desc: "Error occurred in token received callback function"
+  },
+
+  // Configuration Errors
+  invalidCacheLocation: {
+    code: "invalid_cache_location",
+    desc: "The cache contains multiple tokens satisfying the requirements" +
+      "Call AcquireToken again providing more requirements like authority."
+  },
+  noCallback: {
+    code: "no_callback",
+    desc: "Error in configuration: no callback(s) registered for login/acquireTokenRedirect flows." +
+      "Please call handleRedirectCallbacks() with the appropriate callback signatures." +
+      "More information is available here: https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/-basics"
+  },
+  emptyScopes: {
+    code: "empty_input_scopes_error",
+    desc: "Scopes cannot be passed as empty array"
+  },
+  nonArrayScopes: {
+    code: "nonarray_input_scopes_error",
+    desc: "Scopes cannot be passed as non-array"
+  },
+  clientScope: {
+    code: "clientid_input_scopes_error",
+    desc: "Client ID can only be provided as a single scope"
+  },
+  invalidAuthorityType: {
+    code: "invalid_authority_type",
+    desc: "The given authority is not a valid type of authority supported by MSAL. Please see here for valid authorities: <insert URL here>"
+  },
+  authorityUriInsecure: {
+    code: "authority_uri_insecure",
+    desc: "Authority URIs must use https."
+  },
+  authorityUriInvalidPath: {
+    code: "authority_uri_invalid_path",
+    desc: "Given authority URI is invalid."
+  },
+  unsupportedAuthorityValidation: {
+    code: "unsupported_authority_validation",
+    desc: "The authority validation is not supported for this authority type."
+  },
+  b2cAuthorityUriInvalidPath: {
+    code: "b2c_authority_uri_invalid_path",
+    desc: "The given URI for the B2C authority is invalid."
+  },
+
+  // Server Authentication Errors
+  serverUnavailable: {
+    code: "server_unavailable",
+    desc: "Server is temporarily unavailable"
+  },
+  unknownServerError: {
+    code: "unknown_server_error",
+    desc: "Unknown server error"
+  },
+
+  // Interaction Required Errors
+  loginRequired: "login_required",
```

Comment:
everything in here is an object, why aren't these

Reply:
The detailed errors are sent from the server, just need these as constants to be able to detect and throw the correct error type.

Reply:
I think for consistency its better to have an object with a code but not message

Reply:
ok I can do that.